trpcultivate_phenotypes/tests/src/Kernel/Validators/ValidatorBaseTest.php

```diff
@@ -312,4 +325,93 @@ public function testValidatorValidateMethods() {
       "We did not get the exception message we expected when calling BasicallyBase::validate()"
     );
   }
+
+  /**
+   * Test line or row split method.
+   */
+  public function testSplitRowIntoColumns() {
+    $configuration = [];
+    $validator_id = 'fake_basically_base';
+    $plugin_definition = [
+      'id' => $validator_id,
+      'validator_name' => 'Basically Base Validator',
+      'input_types' => ['header-row', 'data-row'],
+    ];
+    $instance = new BasicallyBase($configuration, $validator_id, $plugin_definition);
+    $this->assertIsObject(
+      $instance,
+      "Unable to create fake_basically_base validator instance to test the base class."
+    );
+
+    // Create a data row.
+    // This line captures data values with single/double quotes and leading/trailing spaces.
+    $line = $line_copy = ['Value A', 'Value "B"', 'Value \'C\'', 'Value D ', ' Value E', ' Value F ', ' Value G           '];
```

Comment:
This test would be easier to follow if you called this `$good_line` and `$messy_line` or something similar; `$line_copy` is not very informative.

Reply:
Went with good line and raw line 👍